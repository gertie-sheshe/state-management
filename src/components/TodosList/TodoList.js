import React from "react";
// import { useMutation, useQueryClient } from "react-query";

function TodoList({ error, isLoading, data }) {
  if (error) {
    return <p>Error: {error}</p>;
  }

  if (isLoading) {
    return <p>Loading...</p>;
  }

  const handleOnChange = (e) => {
    e.preventDefault();
    console.log();
  };

  return (
    <form>
      <ul style={{ listStyleType: "none" }}>
        {data?.map((todo) => (
          <li key={todo.id}>
            <input
              onChange={handleOnChange}
              id={todo.id}
              type="checkbox"
              checked={todo.completed}
            />
            <label htmlFor={todo.id}>{todo.title}</label>
          </li>
        ))}
      </ul>
    </form>
  );
}

export default TodoList;
