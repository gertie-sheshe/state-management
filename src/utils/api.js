const API_URL = "http://localhost:3000";
const HEADERS = {
  Accept: "application/json",
  "Content-Type": "application/json",
};

export const apiRequest = async (url, options) => {
  const result = await fetch(url, options);
  const data = await result.json();

  return data;
};

export const getUsers = () => {
  return apiRequest(`${API_URL}/users`);
};

export const getTodos = () => {
  return apiRequest(`${API_URL}/todos`);
};

export const getUserTodos = (userId) => {
  return apiRequest(`${API_URL}/users/${userId}/todos`);
};

export const updateTodo = (todoId, data) => {
  return apiRequest(`${API_URL}/todos/${todoId}`, {
    method: "PUT",
    headers: HEADERS,
    body: JSON.stringify(data),
  });
};

export const createTodo = (data, userId) => {
  return apiRequest(`${API_URL}/users/${userId}/todos`, {
    method: "POST",
    headers: HEADERS,
    body: JSON.stringify(data),
  });
};
