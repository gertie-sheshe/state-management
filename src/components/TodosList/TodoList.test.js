import { render } from "../../tests/utils";
import TodoList from "./TodoList";

describe("TodoList", () => {
  const stubbedTodos = {
    error: false,
    isLoading: false,
    data: [
      {
        userId: 1,
        id: 13,
        title: "et doloremque nulla",
        completed: false,
      },
      {
        title: "home alone",
        completed: false,
        userId: "1",
        id: 14,
      },
    ],
  };
  it("matches snapshot", () => {
    const { asFragment } = render(<TodoList {...stubbedTodos} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
