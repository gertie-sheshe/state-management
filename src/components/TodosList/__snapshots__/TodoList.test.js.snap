// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <ul
      style="list-style-type: none;"
    >
      <li>
        <input
          id="13"
          type="checkbox"
        />
        <label
          for="13"
        >
          et doloremque nulla
        </label>
      </li>
      <li>
        <input
          id="14"
          type="checkbox"
        />
        <label
          for="14"
        >
          home alone
        </label>
      </li>
    </ul>
  </form>
</DocumentFragment>
`;
